### PR TITLE
Server

### PR DESCRIPTION
**Pull request by fl0rek**

### CODE REVIEW

**Review by izaya0 — COMMENTED** [review]
ok

**Review by izaya0 — APPROVED** [review]
ok

File: include/tags.h [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+		// 	game -> player client
+		// 	player client -> server
+		// 	server -> game
+		// 	server -> client
+		// 	client -> game
+		constexpr unsigned char step[] = { 0x12, 0x00, 0x00, 0x03 };
+		// validity of last sent step
+		// valid usages:
+		// 	game -> server
+		constexpr unsigned char step_response[] = { 0x12, 0x00, 0x00, 0x03 };
```
Comment (ASobolewski) [inline include/tags.h:52]: should step_response and step tags be the same?

**Comment by fl0rek** [comment]
Nope, good catch.

File: libnet/include/client.h [hunk 2]
```diff
@@ -48,5 +51,18 @@ bool libnet_send(unsigned char tag, size_t length, unsigned char *value) __attri
  * @param length
  * 	provided buffer length
  */
-ssize_t libnet_wait_for_tag(unsigned char tag, char *buffer, size_t length) __attribute__((warn_unused_result));
+ssize_t libnet_wait_for_tag(const unsigned char tag, unsigned char *buffer,
```
Comment (ASobolewski) [inline libnet/include/client.h:54]: i think it should have block argument, just like in server

File: ServerAPP/server.cc [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+		}
+		written += w;
+	}
+
+	{
+		unsigned char tag[4];
+		read(game_handle.second, tag, 4); //TODO real reading
+		unsigned char tag_length[1];
+		read(game_handle.second, tag_length, 1);
+		unsigned char flipping_flag[1];
```
Comment (izaya0) [inline ServerAPP/server.cc:173]: flag is before length

**Pull request merged**